### PR TITLE
fix: Throwing error 'Failed to execute function' when JSobject is deleted

### LINKED ISSUE #23494: [Bug]: Failed to excute function when JSboject is deleted
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

On page load function is added on page, Delete the JSobject after single run, Navigate from one page to another error is displayed that Filed to execute function - 

https://github.com/appsmithorg/appsmith/assets/80756091/8dc04bf5-6a7a-4db8-b93b-886961bd25a1

### Steps To Reproduce

1. Add app
2. Add below code with setting as "Onpage load" = Yes
3. Create another Page and copy the JSobject from one page to another, Update the setting as Onpage load Yes
4. Verify Function runs on Page load
5. delete the copied JSobject and Navigate from One page to another - error is displayed 

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

cloud

### PR DESCRIPTION
**Pull request by Druthi**
## Description
Throwing error 'Failed to execute function' when JSobject is deleted

#### PR fixes following issue(s)
Fixes #23494 and #21189

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x] Manual

>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [x] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/sagas/ActionExecution/PluginActionSaga.ts [hunk 1]
```diff
@@ -968,6 +969,22 @@ function* executeOnPageLoadJSAction(pageAction: PageAction) {
...
       collectionId,
     );
+
+    if (!collection) {
+      Sentry.captureException(
+        new Error(
+          "Collection present in layoutOnLoadActions but no collection exists ",
+        ),
+        {
+          extra: {
```
Comment (arunvjn) [inline app/client/src/sagas/ActionExecution/PluginActionSaga.ts:979]: Could you also log the `pageId` and `actionId` ?

Comment (Druthi) [reply app/client/src/sagas/ActionExecution/PluginActionSaga.ts:981]: @arunvjn I have updated it as well as added another change according to the comments [here](https://github.com/appsmithorg/appsmith/issues/23494#issuecomment-1576226813)

**Review by arunvjn — DISMISSED** [review]
Looks good!

**Pull request merged**